dvc/stage/run.py

```diff
@@ -56,6 +56,8 @@ def prepare_kwargs(stage, checkpoint_func=None):
         # indicate that checkpoint cmd is being run inside DVC
         kwargs["env"].update(_checkpoint_env(stage))
 
+    kwargs["env"].update(stage.environment)
```

Comment:
@pmrowla 
Introducing `checkpoints` and `dvclive` requires us to set `env` for particular stages.
It seems to me that it worth handling on `stage` level, and might help us if we ever decide to support user-defined variables in stages. If we decide to go that way, I can move `_checkpoint_env` to `stage`.
Related: #2042 #1416 #3633